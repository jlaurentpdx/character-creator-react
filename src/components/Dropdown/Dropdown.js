import './Dropdown.css';
import { useState } from 'react';

export default function Feature({ name, options }) {
  const defaultOption = options.find((item) => item === []);
  const [selection, setSelection] = useState(defaultOption);

  return (
    <label>
      {name}
      <select value={selection} onChange={(e) => setSelection(e.target.value)}>
        {options.map((item) => (
          <option key={item} value={item}>
            {item}
          </option>
        ))}
      </select>
    </label>
  );
}
