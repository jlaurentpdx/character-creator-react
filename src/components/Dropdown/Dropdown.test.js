import { render } from '@testing-library/react';
import Dropdown from './Dropdown';

test('renders a Feature called Head', () => {
  const options = ['bird', 'dog', 'duck', 'horse'];
  const container = render(<Dropdown name="head" options={options} />);
  expect(container).toMatchSnapshot();
});
