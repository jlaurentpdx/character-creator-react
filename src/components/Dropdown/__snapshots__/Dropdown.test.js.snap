// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a Feature called Head 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <label>
        head
        <select>
          <option
            value="bird"
          >
            bird
          </option>
          <option
            value="dog"
          >
            dog
          </option>
          <option
            value="duck"
          >
            duck
          </option>
          <option
            value="horse"
          >
            horse
          </option>
        </select>
      </label>
    </div>
  </body>,
  "container": <div>
    <label>
      head
      <select>
        <option
          value="bird"
        >
          bird
        </option>
        <option
          value="dog"
        >
          dog
        </option>
        <option
          value="duck"
        >
          duck
        </option>
        <option
          value="horse"
        >
          horse
        </option>
      </select>
    </label>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
